### ISSUE TITLE
move from rest to graphql

### ISSUE THREAD

**Issue by DGKSK8LIFE** [projectunic0rn/pub-api#101]

**Comment by roymoran** [comment]
@DGKSK8LIFE how much effort is this? to start I might just explore gaphql for new services that get written.

**Comment by DGKSK8LIFE** [comment]
> @DGKSK8LIFE how much effort is this? to start I might just explore gaphql for new services that get written.

Well, you have to primarily get accustomed to a new way of thinking. The way graphql works is much different than rest. With graphql, you serve all your data under one route, and it's always generated in a POST request, the reason for this is because the client sends query schema instead of filtering through all the data like rest. One thing you'll have to get used to in Graphql as well, is that HTTP methods other than `POST`  aren't used; and url arguments + query strings don't matter anymore. 

SO TL;DR
It may be harder to get the hang of, at first, but the reward of having graphql implemented certainly outweighs the challenge.
